### ISSUE TITLE
PlotlyJS doesn't work as a backend for 0.7

### ISSUE THREAD

**Issue by lewisl** [JuliaPlots/Plots.jl#1728]
The PlotlyJS folks have gotten it to build, pre-compile, and run on 0.7.

The two rather glaring problems are: 
- no series appear on the grid/axes of the plot.
- the plot window cannot be closed.

I reported this at PlotlyJS and Tom replied with some good suggestions. Unfortunately they didn't work.

Spencer replied that it was Plots fault and I should use the PlotlyJS direct API, which is very inconvenient, but it is what is--we can get used to anything.  Unfortunately, it has no mechanism whatever for close--holdover from web and Jupyter focus perhaps (and understandable whenever the browser is the drawing target).

I really don't want to get in the middle here.  There are other backends that work awesomely.  I sort of liked the visuals and functionality of Plotly, but I am quite happy with GR and PyPlot.  I love the ease of packaging a lot of "looks" with PlotThemes.

I can see how some package authors don't like being "wrapped"--up to them.  Maybe it's just time to deprecate Plotly as a backend.  I see lots of helpful posts from Steven and Josef.  There is a good team bring plotting to Julia.

**Comment by sglyon** [comment]
Please (_please_) be patient.

PlotlyJS.jl released a working version for Julia 0.7 less than 24 hours ago.

We will certainly fix this, but haven't yet had time.

There are many people committed to making things work, but things take time.

**Comment by daschw** [comment]
A fix is on the way.

**Comment by sglyon** [comment]
I took a look at fixing things last night and noticed that the javascript code created/injected by WebIO for plots created from this package looked super funky. I have no idea why that would happen.

@daschw happy to chat/pitch in if I can be useful

**Comment by daschw** [comment]
I hope this will be fixed by https://github.com/JuliaPlots/Plots.jl/pull/1730. It works for me locally on julia 1.0

**Comment by sglyon** [comment]
Awesome I'll check it out tonight and give feedback. 

You shouldn't need to install blink as it is required by plotyjs

**Comment by daschw** [comment]
To run ´import Blink; Blink.AtomShell.install()` the user has to install Blink explicitly. If I only have Blink installed as a dependency of PlotlyJS, I get:
```julia
julia> import Blink
ERROR: ArgumentError: Package Blink not found in current path:
- Run `Pkg.add("Blink")` to install the Blink package.
```

**Comment by sglyon** [comment]
Ahh yes that’s right.  Good point. 

I don’t know if that is necessary any more though as I believe the `deps/build.jl`File in Blink will auto install electron (meaning it handles that AtomShell install stuff now)

**Comment by daschw** [comment]
Ah, OK, good to know. I'll try and update the instructions, if that is no longer required.

**Comment by lewisl** [comment]
Sorry, I am posting on wrong issue:

Latest #master, close(p) works but the plot object must always be assigned.  There is no “stack” of windows.  This is OK as different systems work differently.  Do the right thing and the right thing works.

PlotlyJS is only partly working as a Plots backend.  It loads and runs. A plot window is created with axes and grid, but no data series plots.  Closeall() closes the electron window.

I did not test whether Blink automagically installs ElectronWindow as I had done the manual install step.

**Comment by lewisl** [comment]
Update:  there was a no build of PlotlyJS yesterday.  Problems are pretty much the same.  It's OK if this takes some time.

Should I leave this for Plots?

Latest release on #master:
- builds
- compiles
- runs, but with errors

Attempt:
```
using Plots
plotlyjs()
using PlotThemes
theme(:ggplot2)
scatter(randn(200))
```
Electron plot window opens, grid and axes draw. The series as scatter points does not draw. The ggplot2 them (gray background, white gridlines) does not draw--get PlotlyJS defaults.
```
closeall()
```
Successfully closes plot window.

Then:
```
plot(1:4)
ERROR: type SyncPlot has no field view
Stacktrace:
 [1] getproperty(::Any, ::Symbol) at ./sysimg.jl:18
 [2] _create_backend_figure(::Plots.Plot{Plots.PlotlyJSBackend}) at /Users/lewis/.julia/packages/Plots/ex9On/src/backends/plotlyjs.jl:15
 [3] #invokelatest#1 at ./essentials.jl:691 [inlined]
 [4] invokelatest at ./essentials.jl:690 [inlined]
 [5] _plot_setup(::Plots.Plot{Plots.PlotlyJSBackend}, ::Dict{Symbol,Any}, ::Array{Dict{Symbol,Any},1}) at /Users/lewis/.julia/packages/Plots/ex9On/src/pipeline.jl:215
 [6] _plot!(::Plots.Plot{Plots.PlotlyJSBackend}, ::Dict{Symbol,Any}, ::Tuple{UnitRange{Int64}}) at ./logging.jl:313
 [7] #plot#136(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::UnitRange{Int64}) at /Users/lewis/.julia/packages/Plots/ex9On/src/plot.jl:57
 [8] plot(::UnitRange{Int64}) at /Users/lewis/.julia/packages/Plots/ex9On/src/plot.jl:51
 [9] top-level scope at none:0
```
Happy to report over at Plots if that would be better. Progress is certainly being made. Thanks.  (sglyon requested cross-posting)

Everyone is making progress.  We have gr and pyplots;   Plotly has some different features.

**Comment by daschw** [comment]
No, not really. I got it working in Atom (without interactivity) in https://github.com/JuliaPlots/Plots.jl/pull/1730, but Spencer pointed out that this was not working in the REPL.

**Comment by daschw** [comment]
This is not really fixed, see conversation in #1730.

**Comment by daschw** [comment]
Closing as duplicate of #1721
